src/stactools/osc_hazard/cubify.py

```diff
@@ -0,0 +1,141 @@
+import datetime
+import re
+import shutil
+from os import PathLike
+
+import numpy as np
+import xarray as xr
+import zarr
+from hazard.sources.osc_zarr import OscZarr
+
+from .constants import ATTRS
+
+
+def _parse_index_name(key: str, prefix: str = ".*"):
+    match = re.match(f"{prefix}_([0-9]+)c_([a-zA-Z0-9]+)_([a-zA-Z0-9]+)_(\d+)*", key)
```

Comment:
```suggestion
    match = re.match(f"{prefix}_([0-9]+)c_([a-zA-Z0-9-]+)_([a-zA-Z0-9]+)_(\d+)*", key)
```

The gcm for CMIP6 contains a `-`